screen-loans/src/main/java/com/ivy/loans/loan/LoanViewModel.kt

```diff
@@ -236,66 +236,57 @@ class LoanViewModel @Inject constructor(
+                createLoan(event.createLoanData)
+            }
 
-                is LoanScreenEvent.OnAddLoan -> {
-                    _state.value = _state.value.copy(
-                        loanModalData = LoanModalData(
-                            loan = null,
-                            baseCurrency = baseCurrencyCode.value,
-                            selectedAccount = selectedAccount.value
-                        )
-                    )
-                }
+            is LoanScreenEvent.OnAddLoan -> {
+                loanModalData.value = LoanModalData(
+                    loan = null,
+                    baseCurrency = baseCurrencyCode.value,
+                    selectedAccount = selectedAccount.value
+                )
+            }
 
-                is LoanScreenEvent.OnLoanModalDismiss -> {
-                    _state.value = _state.value.copy(
-                        loanModalData = null
-                    )
-                }
+            is LoanScreenEvent.OnLoanModalDismiss -> {
+                loanModalData.value = null
+            }
 
-                is LoanScreenEvent.OnReOrderModalShow -> {
-                    _state.value = _state.value.copy(
-                        reorderModalVisible = event.show
-                    )
-                }
+            is LoanScreenEvent.OnReOrderModalShow -> {
+                reorderModalVisible.value = event.show
+            }
 
-                is LoanScreenEvent.OnReordered -> {
-                    reorder(event.reorderedList)
-                    _state.value = _state.value.copy(
-                        loans = event.reorderedList
-                    )
-                }
+            is LoanScreenEvent.OnReordered -> {
+                reorder(event.reorderedList)
+            }
 
-                is LoanScreenEvent.OnCreateAccount -> {
-                    createAccount(event.accountData)
-                }
+            is LoanScreenEvent.OnCreateAccount -> {
+                createAccount(event.accountData)
             }
         }
     }
-}
-
-data class LoanScreenState(
-    val baseCurrency: String = "",
-    val loans: List<DisplayLoan> = emptyList(),
-    val accounts: List<Account> = emptyList(),
-    val selectedAccount: Account? = null,
-    val loanModalData: LoanModalData? = null,
-    val reorderModalVisible: Boolean = false,
-    val totalOweAmount: String = "",
-    val totalOwedAmount: String = ""
-)
-
-sealed class LoanScreenEvent {
-    data class OnLoanCreate(val createLoanData: CreateLoanData) : LoanScreenEvent()
-    data class OnReordered(val reorderedList: List<DisplayLoan>) : LoanScreenEvent()
-    data class OnCreateAccount(val accountData: CreateAccountData) : LoanScreenEvent()
-    data class OnReOrderModalShow(val show: Boolean) : LoanScreenEvent()
-    object OnAddLoan : LoanScreenEvent()
-    object OnLoanModalDismiss : LoanScreenEvent()
-}
+
+    @Composable
+    private fun getReorderModalVisible() = reorderModalVisible.value
```

Comment:
These getters should be right below `@Composable uiState()`. Functions order:
- uiState()
- getters
- onEvent
- private event functions

Reply:
Sure, is this something that we are following in general or are there some guidelines? Please share resource for same.

Reply:
No resource, unfortunately. It's common formatting technique. The main idea is:
- functions that are called by other functions should be exactly below them
- uiState() calls the getters 

It's a common practice and an idea from the Clean Code book that I don't like much 😀

Reply:
The best course of action would be to enforce that using Detekt, if technically possible

Reply:
Got it, makes sense. Will result in easier code readability if being followed all across.

Reply:
Exactly, that's the idea 💯 Makes easy to understand the code w/o scrolling like crazy

Reply:
Will look if we can do it using detekt.